node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -730,9 +769,18 @@ async fn handle_peer_view_change(state: &mut State, peer_id: PeerId, view: View)
 	let peer_data = state.peer_data.entry(peer_id).or_default();
 
 	peer_data.update_view(view);
-	state
-		.requested_collations
-		.retain(|pc, _| pc.peer_id != peer_id || peer_data.has_advertised(&pc.relay_parent));
+
+	let retain_predicate =
+		|pc: &PendingCollation| pc.peer_id != peer_id || peer_data.has_advertised(&pc.relay_parent);
+	// We iterate twice as `Hashmap::drain_filter` is not yet stable. The alternative would be
+	// to store the `keys()` and mutate in a single `for` pass but would require extra space.
+	state.collation_requests_cancel_handles.iter().for_each(|(pc, handle)| {
+		if !(retain_predicate)(pc) {
+			handle.cancel();
+		}
+	});
```

Comment:
You don't need `drain_filter`, `retain` takes arguments mutably when possible.

```suggestion
	let retain_predicate =
		|pc: &PendingCollation| pc.peer_id != peer_id || peer_data.has_advertised(&pc.relay_parent);
	state.collation_requests_cancel_handles.retain(|pc, handle| {
		let keep = (retain_predicate)(pc);
		if !keep {
			handle.cancel();
		}
		keep
	});
```

Reply:
cool, thanks for the suggestion